markets/perps-market/contracts/storage/GlobalPerpsMarket.sol

```diff
@@ -47,16 +53,28 @@ library GlobalPerpsMarket {
         uint128 synthMarketId,
         int synthAmount
     ) internal {
+        uint collateralAmount = self.collateralAmounts[synthMarketId];
         if (synthAmount > 0) {
-            if (
-                self.collateralAmounts[synthMarketId] + synthAmount.toUint() >
-                GlobalPerpsMarketConfiguration.load().maxCollateralAmounts[synthMarketId]
-            ) {
-                revert MaxCollateralExceeded(synthMarketId);
+            uint maxAmount = GlobalPerpsMarketConfiguration.load().maxCollateralAmounts[
+                synthMarketId
+            ];
+            uint newCollateralAmount = collateralAmount + synthAmount.toUint();
+            if (newCollateralAmount > maxAmount) {
+                revert MaxCollateralExceeded(
+                    synthMarketId,
+                    maxAmount,
+                    collateralAmount,
+                    synthAmount.toUint()
+                );
             } else {
                 self.collateralAmounts[synthMarketId] += synthAmount.toUint();
             }
         } else {
+            uint synthAmountAbs = MathUtil.abs(synthAmount);
+            if (collateralAmount < synthAmountAbs) {
+                revert InsufficientCollateral(synthMarketId, collateralAmount, synthAmountAbs);
+            }
```

Comment:
makes sense

Reply:
I noticed we have `InsufficientMargin` in the AsyncOrder file. Do we have a definition on the differences between margin and collateral? And do we want to refuse errors as much as possible?

Reply:
Hmm we also have a `error InsufficientMarginError(uint leftover);` in PerpsAccount.

Reply:
we should distinguish them in this way imo:

- `InvalidWithdrawalAmount(amountAvailableToWithdraw, amountRequested)`: this one for the above scenario where trader is attempting to withdraw more than the entire market has.
- `InsufficientMargin()`: currently this is thrown when validating orders.  I have a TODO to create separate errors for which this is being reused during validation.
- `InsufficientCollateral()`: this should occur very rarely cuz by the time you're deducting, you've already checked you have enough to cover fees.  so yeah lets rename `InsufficientMarginError` to this or something similar?

all these are suggestions; feel free to suggest other names and ways to distinguish 👍🏽